### ISSUE TITLE
epub: Crash on certain epub files

### ISSUE THREAD

**Issue by infirit** [mate-desktop/atril#97]
On some epub file I get a crash in libepubdocumen.

Backtrace:
```
Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff77fff700 (LWP 4725)]
0x00007fff8c4f98d3 in page_set_function ()
   from /usr/lib64/atril/3/backends/libepubdocument.so
#0  0x00007fff8c4f98d3 in page_set_function ()
   from /usr/lib64/atril/3/backends/libepubdocument.so
#1  0x00007ffff509d178 in g_list_foreach (list=<optimized out>, 
    func=0x7fff8c4f98c0 <page_set_function>, user_data=0x7a95a0)
    at /var/tmp/portage/dev-libs/glib-2.40.2/work/glib-2.40.2/glib/glist.c:993
#2  0x00007fff8c4fb9a8 in epub_document_load ()
   from /usr/lib64/atril/3/backends/libepubdocument.so
#3  0x00007ffff7797476 in ev_document_load ()
   from /usr/lib64/libatrildocument.so.3
#4  0x00007ffff7798f6b in ev_document_factory_get_document ()
   from /usr/lib64/libatrildocument.so.3
#5  0x00007ffff7559fff in ev_job_load_run () from /usr/lib64/libatrilview.so.3
#6  0x00007ffff755af40 in ev_job_thread_proxy ()
   from /usr/lib64/libatrilview.so.3
#7  0x00007ffff50c58f5 in g_thread_proxy (data=0xb9c190)
    at /var/tmp/portage/dev-libs/glib-2.40.2/work/glib-2.40.2/glib/gthread.c:764
#8  0x00007ffff4b43083 in start_thread (arg=0x7fff77fff700)
    at pthread_create.c:309
#9  0x00007ffff488003d in clone ()
    at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
```

Example file: https://www.dropbox.com/s/p5ytz5tk4tde0n8/Do%20Plants%20Think_%20-%20Scientific%20American.epub?dl=0

**Comment by monsta** [comment]
I guess it can be considered fixed? At least the mentioned document doesn't cause the crash for me.

**Comment by ZrN-kAyDqY** [comment]
Atril 1.10.2:   The book has no errors according to Calibre's ebook editor
@infirit's epub crashes it as well.

$ atril a.epub
No bp log location saved, using default.
Vector smash protection is enabled.
No bp log location saved, using default.
[000:000] Browser XEmbed support present: 1
[000:000] Browser toolkit is Gtk2.
[000:000] Using Gtk2 toolkit
[000:011] Warning(optionsfile.cc:30): Load: Could not open file, err=2
[000:011] No bp log location saved, using default.
[000:001] Warning(optionsfile.cc:30): Load: Could not open file, err=2
[000:001] No bp log location saved, using default.
[000:002] Browser XEmbed support present: 1
[000:002] Browser toolkit is Gtk2.
[000:002] Using Gtk2 toolkit
Segmentation fault (core dumped)

**Comment by monsta** [comment]
> The book has no errors according to Calibre's ebook editor

Which book?

**Comment by ZrN-kAyDqY** [comment]
The book I tested with initially had DRM, so I removed it to test it and view (tested it without fixing the warnings/errors Calibre found, and after fixing them).  That book I'm definitely not posting a link to.  However I did get the same errors on both that book and the "Do Plants Think" book posted earlier in this thread (errors and Atril crashing).

Here's another one that crashes the same and is public domain (use epub with no images:
http://www.gutenberg.org/ebooks/226?msg=welcome_stranger
